client/my-sites/patterns/index.web.ts

```diff
@@ -1,17 +1,21 @@
 import { getLanguageRouteParam } from '@automattic/i18n-utils';
-import { makeLayout, redirectWithoutLocaleParamInFrontIfLoggedIn } from 'calypso/controller';
-import { setLocaleMiddleware } from 'calypso/controller/shared';
+import {
+	clientRouter,
+	makeLayout,
+	redirectWithoutLocaleParamInFrontIfLoggedIn,
+	render as clientRender,
+} from 'calypso/controller';
 import { fetchPatterns, renderPatterns } from 'calypso/my-sites/patterns/controller';
 
-export default function ( router ) {
+export default function ( router: typeof clientRouter ) {
 	const langParam = getLanguageRouteParam();
 
 	router(
 		[ '/patterns', `/${ langParam }/patterns` ],
 		redirectWithoutLocaleParamInFrontIfLoggedIn,
-		setLocaleMiddleware(),
```

Comment:
I've removed this middleware since it should be useless because of the one just above (which removes the `:lang` parameter from the path).